04_pig_latin/pig_latin.rb

```diff
@@ -1 +1,29 @@
 #write your code here
+# method takes a string, checks if the first letter of the word is a vowel, if so it adds 'ay'
+# if the word begins with a consonant, then it finds the first vowel, takes the letters before that
+# and adds them to the end of the word, then adds 'ay', finally all words joined with a space between
+def translate(str)
+  vowels = ['a','e','i','o','u']
+  words = str.split
+  translated_words = words.map do |word|
+    if vowels.include?(word[0])
+      word + 'ay'
+    else
+      # handles the case where the first syllable contains qu, either at the beginning or with consonant
+      # in front
+      if word.start_with?('qu')
+        word[2..-1] + word[0..1] + 'ay'
+      elsif word[1..2] == 'qu'
+        word[3..-1] + word[0..2] + 'ay'
+      else
+        first_vowel = word.index(/[aeiou]/)
+        word[first_vowel..-1] + word[0...first_vowel] + 'ay'
+      end        
+    end
+  end
+  translated_words.join(' ')  
+end
```

Comment:
Good. This works too, and imho, this assignment is 👎🏼 .

But, you've used a bit of some [regex](https://en.wikipedia.org/wiki/Regular_expression) syntax here `/aeiou/` and I want to share a solution that simplifies this logic a bit more.

1. You don't need the variable `words`: you're _only_ using it to call `#map` on the next line, so you might as well chain your method calls and avoid declaring a variable.

   You also don't need `translated_words`, because you can chain `#join` at the end of the block:

    ```ruby
    str.split.map do |word|
      # translate block logic
    end.join(' ')
    ```

2. Regex Logic: 

   > NOTE: If this is overwhelming, ignore it!

   You set `vowels` array, but only used it once and then used other ways to determine where vowels were.

   We can simplify this like so:

    ```ruby
    str.split.map do |word|
      # if the `word` matches the given regex 
      # (`=~` is regex comparison operator)
      # `/^[aeiou]/`  means a string that starts with (`^`) any character within the brackets
      if word =~ /^[aeiou]/
        word + "ay"
      else
       # Otherwise, call `#sub` on the `word`
       # (see below for explanation of this regex)
        word.sub(/(^[^aeio]*qu|^[^aeiou]+)(.*)/, '\2\1ay')
      end
    end.join(" ")
   ```

### `#sub` regex explanation

`#sub` takes two arguments, a **pattern** and a **replacement**. They can be plain string or a regex, which they are here.

The **pattern** is:

```ruby
/(^[^aeio]*qu|^[^aeiou]+)(.*)/
```

It will match:
1. Any word who's _first group_(`()`) starts with(`^`) with either:
   1.  a character that is NOT (`[^...]`) `aeio` and is followed by `qu`
    2. (`|`) a character that is NOT ([^...]) `aeiou`
2. And is followed by a _second group_ (`()`) of any character(`.`), any number of times (`*`)
 
The **replacement** is:

```ruby
'\2\1ay'
```
It will **replace** any matches by reordering the string to be _second group_ (`\2`) followed by _first group_ (`\1`) followed by `"ay"`